src/main-process/squirrel-update.js

```diff
@@ -0,0 +1,209 @@
+          fs.writeFile(apmShCommandPath, apmShCommand, () => callback())
+        })
+      })
+    })
+  }
+
+  function addBinToPath (pathSegments, callback) {
+    pathSegments.push(binFolder)
+    const newPathEnv = pathSegments.join(';')
+    return spawnSetx(['Path', newPathEnv], callback)
+  }
+
+  return installCommands((error) => {
+    if (error != null) { return callback(error) }
+
+    return WinPowerShell.getPath((error, pathEnv) => {
+      if (error != null) { return callback(error) }
+
+      const pathSegments = pathEnv.split(/;+/).filter(pathSegment => pathSegment)
+      if (pathSegments.indexOf(binFolder) === -1) {
+        return addBinToPath(pathSegments, callback)
+      } else {
+        return callback()
+      }
+    })
+  })
+}
+
+// Remove atom and apm from the PATH
+function removeCommandsFromPath (callback) {
+  WinPowerShell.getPath((error, pathEnv) => {
+    if (error != null) { return callback(error) }
+
+    const pathSegments = pathEnv.split(/;+/).filter(pathSegment => pathSegment && (pathSegment !== binFolder))
+    const newPathEnv = pathSegments.join(';')
+
+    if (pathEnv !== newPathEnv) {
+      return spawnSetx(['Path', newPathEnv], callback)
+    } else {
+      return callback()
+    }
+  })
+}
+
+// Create a desktop and start menu shortcut by using the command line API
+// provided by Squirrel's Update.exe
+function createShortcuts (callback) {
+  spawnUpdate(['--createShortcut', exeName], callback)
+}
+
+// Update the desktop and start menu shortcuts by using the command line API
+// provided by Squirrel's Update.exe
+function updateShortcuts (callback) {
+  const homeDirectory = fs.getHomeDirectory()
+
+  if (homeDirectory) {
+    const desktopShortcutPath = path.join(homeDirectory, 'Desktop', 'Atom.lnk')
+    // Check if the desktop shortcut has been previously deleted and
+    // and keep it deleted if it was
+    return fs.exists(desktopShortcutPath, desktopShortcutExists =>
+      createShortcuts(() => {
+        if (desktopShortcutExists) {
+          return callback()
+        } else {
+          // Remove the unwanted desktop shortcut that was recreated
+          return fs.unlink(desktopShortcutPath, callback)
+        }
+      })
+    )
+  } else {
+    return createShortcuts(callback)
+  }
+}
+
+// Remove the desktop and start menu shortcuts by using the command line API
+// provided by Squirrel's Update.exe
+function removeShortcuts (callback) {
+  spawnUpdate(['--removeShortcut', exeName], callback)
+}
+
+function updateContextMenus (callback) {
+  WinShell.fileContextMenu.update(() => {
+    WinShell.folderContextMenu.update(() => {
+      WinShell.folderBackgroundContextMenu.update(() => callback())
+    })
+  })
+}
+
+export { spawnUpdate as spawn }
+
+// Is the Update.exe installed with Atom?
+export function existsSync () {
+  return fs.existsSync(updateDotExe)
+}
+
+// Restart Atom using the version pointed to by the atom.cmd shim
+export function restartAtom (app) {
```

Comment:
Node is doing it too, though it is CJS.
https://github.com/nodejs/node/pull/12654
https://github.com/nodejs/node/pull/12755

This maybe a matter of taste, but I'd say that exports-last style allows you to figure out what's happening in file first, and then just look at what part of that API is exported.